File: disparity_wrap/wrap.cpp
#include"wrap.h"
using namespace std;


void wrap(int_fixed img[x*y], char disp[x*y], int_fixed dst[x*y]){
    for (int i =0; i<x ;i++)
    {
        for(int j=0;j<y ;j++){
            int d = disp[i + x*j];
            //move the current pixel to the cordinates of the correspending pixel
            if ( i+d < x) dst[i+x*j] = img[i+d + x*j];
            //for the pixels with no correspending pixels in range 
            else dst[i+x*j] = 255;
        }
    }
}

File: 2_disparity_wrap/wrap.py
import cv2 
import numpy as np


def wrap(img,disp):
    dst = img
    for i in range(80):
        for j in range(80):
            d = disp[i][j]
            if ( j + d < 80)  :
                dst[i][j] = img[i][j+d]
            else :  dst[i][j] = 255
    return dst


if __name__ == "__main__":
    image = cv2.imread("data1/1_right.pgm")
    disparity = cv2.imread("data1/0_disparity_map.pgm",0)
    cv2.imshow('image',image)
    wrapped =wrap(image,disparity)
    cv2.waitKey(0)
    
